### DISCUSSION TITLE
modify recording data to include other objects

### DISCUSSION THREAD

**Discussion by Zzerr0r** [microsoft/AirSim#4592] [Support Q&A]
I want to edit the MultirotorPawnSimApi.cpp or PawnSimApi.cpp in such a way that it also records the position of another object. 
I would have done it manually with the pythonClient via client.simGetObjectPose() but then i would have different timestamps than in the recording. I Also couldn't manage to get recording all data manually to work because whenever I call client.simGetImages() my simulation freezes and that messes with the timestamps as well.

**Comment by alonfaraj** [comment]
Hi @Zzerr0r,
If you build airsim from source you can modify the recording feature to fits your needs.   
You can find more details here https://microsoft.github.io/AirSim/modify_recording_data/

**Reply by Zzerr0r** [reply]
Yes I know, i already modified it to include the distance sensor. I was wondering how i could get acces to the method simGetObjectPose() from within the source code

**Reply by alonfaraj** [reply]
Sorry for the delay,
you can use this inside `MultirotorPawnSimApi.cpp` or anywhere else:
```
const auto simmode = ASimModeBase::getSimMode();
const auto pose = simmode->getApiProvider()->getWorldSimApi()->getObjectPose("Foo");
```
